### PR TITLE
fix: Windows compatibility

### PR DESCRIPTION
**Pull request by luisorbaiceta**
fixes the error described in #215 where the subsections of the index were not expanding

- [x] commit message and code follows the [Developer's Certification of Origin]

### CODE REVIEW

**Comment by luisorbaiceta** [comment]
@lmammino this fixes what we talked about in #215. I have already sent a PR to the contentHash plugin that completes the fix for all the issues when building the website on Windows and as far as I am concerned everything seems to be working fine now. 

I am leaving It as a draft until I check all the codebase for any other path issues regarding the use of '\\'

**Comment by lmammino** [comment]
Hello Luis, this sounds good (and thanks again for your efforts!)

If you don't get any reply on the contenthash plugin, i am ok with "copying it" into this project and doing all the necessary changes locally. If i remember correctly the original plugin isn't particularly big, so I'd say it could make sense to internalise the plugin if it helps us to move forward quickly.

One more small detail: rather than always using '\' for the path, maybe we should be using the specific path separator for the current system.

I think we can do that with the `path` module:



```js
const path = require('path')
console.log(path.sep) //  '/' (on macOs)
```

**Comment by jsumners** [comment]
`path.join` should always be used to join file system path components.

**Comment by luisorbaiceta** [comment]
> If i remember correctly the original plugin isn't particularly big, so I'd say it could make sense to internalise the plugin if it helps us to move forward quickly

That's right. I can include directly in the plugin workflow in this PR. Doesn't seem that we'll be getting any reply soon. @Eomm opened an Issue and there was no reply either.

> One more small detail: rather than always using '' for the path, maybe we should be using the specific path separator for the current system.

Where do you mean?

**Comment by luisorbaiceta** [comment]
> `path.join` should always be used to join file system path components.

Yes. The problem here is we want the contentHash plugin to match routes with `/` (As this is the used by the html files), and `path.join` will still add the windows specific separators

**Comment by lmammino** [comment]
> Yes. The problem here is we want the contentHash plugin to match routes with / (As this is the used by the html files), and path.join will still add the windows specific separators


Yeah, I suppose a clean solution would be that contenthash should store the paths following the current os path separators...

**Comment by luisorbaiceta** [comment]
> Yeah, I suppose a clean solution would be that contenthash should store the paths following the current os path separators...

I think that's what is happening right now, and the reason of the issue. In the end, the renderer is going to look for `{{ hashes['images/bg-pattern-dark.png'] }}` and If Windows was used there is not going to be a valid match. Maybe a valid approach would be to achieve something like `{{ hashes['images${separator}bg-pattern-dark.png'] }}` and pass the separator as a variable with `path.sep`. The path stored in hashes will need to be stored with `/` anyway as it is the only way to be recognized by html... What do you think?

I will see if using some library like [slash](https://github.com/sindresorhus/slash) could give us a hand with this so we can use `path.join` as suggested by @jsumners

**Comment by lmammino** [comment]
> One more small detail: rather than always using '' for the path, maybe we should be using the specific path separator for the current system.
> >Where do you mean?

Lol sorry, the `\` got escaped by markdown. I was just suggesting to use either `\` or `/` depending on the OS. But I think @jsumners 's suggestion to go with `path.join` is certainly a much more reliable and elegant approach!

I didn't know about `slash`. Great finding :)

**Review by mcollina — APPROVED** [review]
lgtm

File: src/website/layouts/docs_page.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {% extends "default.html" %}
 
 {% block content %}
+{% set path = path | replace("\\", "/") %}
```
Comment (lmammino) [inline src/website/layouts/docs_page.html:4]: Do we really need this one?

**Review by lmammino — APPROVED** [review]
This looks good to me but I have one question on that `path` variable in the template... I am unsure why we need to change that one (since we don't use for assets but for hyperlinks)

File: src/website/layouts/docs_page.html (hunk 1 above)
Comment (luisorbaiceta) [reply src/website/layouts/docs_page.html:4]: It is necessary. When fetching the docs page, in order to build the subindex in the left column, if this step is not included:

[This condition will always fail in windows systems](https://github.com/fastify/website/blob/8f53a1bf5ab4c0c3ee1c7fdeaf186b83d521617a/src/website/layouts/docs_page.html#L29)

[This condition too](https://github.com/fastify/website/blob/8f53a1bf5ab4c0c3ee1c7fdeaf186b83d521617a/src/website/layouts/docs_page.html#L30)

**Review by Eomm — APPROVED** [review]
This PR works 👍🏽 

As a side note, Node >= 14 won't work 😅 

```
Error: Missing binding website\node_modules\metalsmith-sass\node_modules\node-sass\vendor\win32-x64-83\binding.node                                                                                             Node Sass could not find a binding for your current environment: Windows 64-bit with Node.js 14.x                                                                                                                                               Found bindings for the following environments:                                                                            - Windows 64-bit with Node.js 12.x                                                                                                                                                                                                            This usually happens because your environment has changed since running `npm install`.                                  Run `npm rebuild node-sass` to download the binding for your current environment.                                           at module.exports (C:\Users\behem\workspaceFastify\website\node_modules\metalsmith-sass\node_modules\node-sass\lib\binding.js:15:13)
```

**Comment by luisorbaiceta** [comment]
> As a side note, Node >= 14 won't work 😅 

I will take a look at this tomorrow and see if I can fix it

**Comment by lmammino** [comment]
> As a side note, Node >= 14 won't work 😅

I think we just need to bump node-sass to v6. But we can do that in a dedicated PR. I am ok to merge this one!

**Pull request merged**